libraries/app/test/app_test.js

```diff
@@ -25,6 +27,19 @@ suite(testing.suiteName(), function() {
               valueSet: req.requestId,
             }));
           });
+          router.get('/__version__', function(req, res) {
+            const taskclusterVersionFile = path.resolve(__dirname, 'fixtures', 'taskcluster-version');
+
+            try {
+              const taskclusterVersion = fs.readFileSync(taskclusterVersionFile).toString().trim();
+
+              res.header('Content-Type', 'text/plain');
+              res.send(taskclusterVersion);
+            } catch (err) {
+              res.header('Content-Type', 'text/plain');
+              res.status(500).send('Could not locate the version file');
+            }
+          });
```

Comment:
The test below tests *this* implementation, and not real the one in `index.js`.

I suspect that the real implementation could be tested using `mock-fs`.  There are some uses of that package in tc-lib-references if you want to check them out.